heareval/tasks/pipeline.py

```diff
@@ -284,18 +317,49 @@ def get_metadata(self):
         )
         return metadata
 
+    def subsample_metadata(self, metadata, max_files):
+
+        # Get group count for each group
+        grp_count = metadata["stratify_key"].value_counts() * max_files / len(metadata)
+        # Sort by the split key and the subsample key
+        metadata = metadata.sort_values(
+            by=["split_key", "subsample_key"], ascending=[True, True]
+        )
+        # Groupby and select the required sample from each group
+        sampled_metadata = pd.concat(
+            [
+                # Ensure at least 1 sample is selected for each group
+                stratify_grp.head(1 if grp_count[grp] < 1 else int(grp_count[grp]))
+                for grp, stratify_grp in metadata.groupby("stratify_key")
+            ]
+        )
```

Comment:
Process - 
- Get the count required for each group.
- Sort the metadata by the `split_key` and the `subsample_key`
- For each group select the count required and concatanate

Reply:
This should be a comment